example-generator/src/main/java/org/ehrbase/openehr/sdk/examplegenerator/ExampleGeneratorConfig.java

```diff
@@ -200,15 +204,19 @@ private static int determineIndex(double t, int size) {
         return Math.min(Math.max(0, (int) (t * size)), size - 1);
     }
 
-    private static Optional<WebTemplateInput> getInput(WebTemplateNode node, String suffix) {
-        return Optional.ofNullable(node)
-                .map(WebTemplateNode::getInputs)
-                .stream()
-                .flatMap(List::stream)
-                .filter(e -> Objects.equals(suffix, e.getSuffix()))
-                .findFirst();
+    private static Optional<WebTemplateInput> getInput(WebTemplateNode node, Predicate<WebTemplateInput> predicate) {
+        return getInputs(node, predicate).stream().findFirst();
     }
-
+    
+    private static List<WebTemplateInput> getInputs(WebTemplateNode node, Predicate<WebTemplateInput> predicate) {
```

Comment:
Do you really need that method.
It is only used once and the predicate is not used.
Effectively it spares you from calling once
```
Optional.ofNullable(node)
  .map(WebTemplateNode::getInputs)
  .stream()
```

Reply:
For me is was more clear to have this method than just inlining